dts/msm8916/msm8916-512mb-mtp.dts

```diff
@@ -10,13 +10,19 @@
 	qcom,msm-id = <206 0>;
 	qcom,board-id = <8 0x100>;
 	
+	// Also supports UFI003_MB_V02 and MF601 variants
 	zhihe-ufi-001c {
 		model = "ufi-001b/ufi-001c 4G Modem Stick";
 		compatible = "zhihe,ufi-001c", "qcom,msm8916", "lk2nd,device";
 
 		/* Register EDL button as HOME button */
-		/* This is the only button on this device */
-		lk2nd,keys = <KEY_HOME 37 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
+		/* This is the only button on UFI-001C */
+		lk2nd,keys = <KEY_HOME 37 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>,
+		// The following keys are found on MF601
+		// RESET button on the back of the device
+			     <KEY_HOME 34 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
+		// WPS button near the power button(may not exist on some variants)
+			     <KEY_HOME 107 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
```

Comment:
Did you make sure these pins are not tied down or connected to something on other boards? This configurations asserts the gpio to sense the button (effectively enabling it) so it might assert something, or, if the pin is tied to ground, force button to be always held

Reply:
> Did you make sure these pins are not tied down or connected to something on other boards? This configurations asserts the gpio to sense the button (effectively enabling it) so it might assert something, or, if the pin is tied to ground, force button to be always held

Tested on UFI-001C.